int32 event_type
int32 angle_old_1
int32 angle_old_2
int32 angle_new_1
int32 angle_new_2